File: ruby/day1_part1_2017.rb

puts File.read("input.txt").strip.chars.map(&:to_i).each_with_index.inject(0) { |sum, (num, index)| num == File.read("input.txt").strip.chars[(index + 1) % File.read("input.txt").strip.length].to_i ? sum + num : sum }

File: day1_part1_2017.py

#!/usr/bin/env python3
with open('input.txt', 'r') as f:
    data = f.read().strip()

sum = 0
for i in range(len(data)):
    if data[i] == data[(i + 1) % len(data)]:
        sum += int(data[i])

print(sum)
